app/views/split_checkout/_your_details.html.haml

```diff
@@ -0,0 +1,112 @@
+  %div.checkout-title
+    = t("split_checkout.step1.billing_address.title")
+
+  %div.checkout-input
+    = fields_for :bill_address, @order.bill_address do |bill_address|
+      = bill_address.label :address1, t("split_checkout.step1.billing_address.address1.label")
+      = bill_address.text_field :address1, { placeholder: t("split_checkout.step1.billing_address.address1.placeholder") }
+    = f.error_message_on "bill_address.address1"
+
+  %div.checkout-input
+    = fields_for :bill_address, @order.bill_address do |bill_address|
+      = f.label :address2, t("split_checkout.step1.billing_address.address2.label")
+      = f.text_field :address2, { placeholder: t("split_checkout.step1.billing_address.address2.placeholder") }
+    = f.error_message_on "bill_address.address2"
+
+  %div.checkout-input
+    = fields_for :bill_address, @order.bill_address do |bill_address|
+      = bill_address.label :city, t("split_checkout.step1.billing_address.city.label")
+      = bill_address.text_field :city, { placeholder: t("split_checkout.step1.billing_address.city.placeholder") }
+    = f.error_message_on "bill_address.city"
+
+  %div.checkout-input
+    = fields_for :bill_address, @order.bill_address do |bill_address|
+      = bill_address.label :state_id, t("split_checkout.step1.billing_address.state_id.label")
+      = bill_address.select :state_id, @countries_with_states, { }, { "data-dependant-select-target": "select" }
+
+  %div.checkout-input
+    = fields_for :bill_address, @order.bill_address do |bill_address|
+      = bill_address.label :zipcode, t("split_checkout.step1.billing_address.zipcode.label")
+      = bill_address.text_field :zipcode, { placeholder: t("split_checkout.step1.billing_address.zipcode.placeholder") }
+    = f.error_message_on "bill_address.zipcode"
+
+  %div.checkout-input
+    = fields_for :bill_address, @order.bill_address do |bill_address|
+      = bill_address.label :country_id, t("split_checkout.step1.billing_address.country_id.label")
+      = bill_address.select :country_id, @countries, { selected: @order.bill_address.country_id || DefaultCountry.id }, {"data-dependant-select-target": "source", "data-action": "dependant-select#handleSelectChange"}
+
+  - if spree_current_user||true
+    %div.checkout-input
+      = f.check_box :checkout_default_bill_address
+      = f.label :checkout_default_bill_address, t(:checkout_default_bill_address)
+
+  %div.checkout-substep{ "data-controller": "toggle" }
+    -# DELIVERY ADDRESS
+    %div.checkout-title
+      = t("split_checkout.step1.delivery_address.title")
+
+    = @shipping_methods.each do |shipping_method| 
+      %div.checkout-input
+        = fields_for shipping_method do |shipping_method_form|
+          = shipping_method_form.radio_button :name, shipping_method.id, { id: "shipping_method_" + shipping_method.id.to_s, "data-description": shipping_method.description, "data-action": "toggle#toggle", "data-toggle-show":  shipping_method.require_ship_address }
```

Comment:
You can add multiple actions in a single `data-action` attribute, so for example here we can use:
```ruby
"data-action": "toggle#toggle splitcheckout#selectShippingMethod"
```
and it'll trigger both :+1:

Reply:
There's a lot going on in this line, I think I would split it like this:
```ruby
= shipping_method_form.radio_button :name, shipping_method.id,
  id: "shipping_method_#{shipping_method.id}",
  "data-description": shipping_method.description,
  "data-action": "toggle#toggle splitcheckout#selectShippingMethod",
  "data-toggle-show": shipping_method.require_ship_address
```